calendar/api/events_test.go

```diff
@@ -0,0 +1,611 @@
+				mockLogger.EXPECT().With(gomock.Any()).Return(mockLoggerWith)
+				mockLoggerWith.EXPECT().Errorf("error sending post to channel about linked event")
+			},
+			assertions: func(rec *httptest.ResponseRecorder) {
+				assert.Equal(t, http.StatusCreated, rec.Result().StatusCode)
+				responseBody, _ := io.ReadAll(rec.Body)
+				assert.Contains(t, string(responseBody), "true")
+			},
+		},
+		{
+			name: "Successfully create event with channelID",
+			setup: func(req *http.Request) {
+				req.Header.Set(MMUserIDHeader, MockUserID)
+				currentTime := time.Now()
+				validJSON := GetCurrentTimeRequestBodyJSON()
+				req.Body = io.NopCloser(bytes.NewBufferString(validJSON))
+				mockOAauthToken := oauth2.Token{}
+				mockStore.EXPECT().LoadUser(MockUserID).Return(&store.User{MattermostUserID: MockUserID, OAuth2Token: &mockOAauthToken, Remote: &remote.User{ID: MockRemoteUserID}}, nil)
+				mockPluginAPI.EXPECT().CanLinkEventToChannel(MockChannelID, MockUserID).Return(true)
+				mockRemote.EXPECT().MakeClient(gomock.Any(), &mockOAauthToken).Return(mockRemoteClient)
+				mockRemoteClient.EXPECT().GetMailboxSettings(MockRemoteUserID).Return(&remote.MailboxSettings{TimeZone: "UTC"}, nil)
+				mockEvent := &remote.Event{
+					Start: &remote.DateTime{
+						DateTime: currentTime.Add(time.Hour).Format("2006-01-02T15:04:05Z"),
+						TimeZone: "UTC",
+					},
+					End: &remote.DateTime{
+						DateTime: currentTime.Add(2 * time.Hour).Format("2006-01-02T15:04:05Z"),
+						TimeZone: "UTC",
+					},
+					Subject:  "Meeting with team",
+					Location: &remote.Location{DisplayName: "Conference Room"},
+					Conference: &remote.Conference{
+						URL:         "https://example.com/conference",
+						Application: "Zoom",
+					},
+					Attendees: []*remote.Attendee{
+						{EmailAddress: &remote.EmailAddress{Name: "John Doe", Address: "john.doe@example.com"}},
+						{EmailAddress: &remote.EmailAddress{Name: "Jane Smith", Address: "jane.smith@example.com"}},
+					},
+				}
+				mockRemoteClient.EXPECT().CreateEvent(MockRemoteUserID, gomock.Any()).Return(mockEvent, nil)
+				mockStore.EXPECT().StoreUserLinkedEvent(MockUserID, gomock.Any(), MockChannelID).Return(nil)
+				mockStore.EXPECT().AddLinkedChannelToEvent(gomock.Any(), MockChannelID).Return(nil)
+				mockPoster.EXPECT().CreatePost(gomock.Any()).Return(nil)
+			},
+			assertions: func(rec *httptest.ResponseRecorder) {
+				assert.Equal(t, http.StatusCreated, rec.Result().StatusCode)
+				responseBody, _ := io.ReadAll(rec.Body)
+				assert.Contains(t, string(responseBody), "true")
+			},
+		},
+		{
+			name: "Event created successfully without channelID",
+			setup: func(req *http.Request) {
+				req.Header.Set(MMUserIDHeader, MockUserID)
+				currentTime := time.Now()
+				date := currentTime.Format("2006-01-02")
+				startTime := currentTime.Add(time.Hour).Format("15:04")
+				endTime := currentTime.Add(2 * time.Hour).Format("15:04")
+				validJSON := fmt.Sprintf(`{
```

Comment:
We have created a constant similar to this. Cant we use that here? Maybe we can create a util which accepts a param channeID